src/pip/_internal/req/req_file.py

```diff
@@ -227,7 +224,7 @@ def handle_option_line(
     # type:  (...) -> None
 
     # percolate hash-checking option upward
-    if opts.require_hashes:
+    if opts.require_hashes and options:
```

Comment:
```suggestion
    if options and opts.require_hashes:
```

I am wondering if putting `options` first is slightly better since it's on the LHS of the following assignment and has to be mandatorily checked.

Reply:
done